csds-client/client/util.go

```diff
@@ -0,0 +1,259 @@
+			break
+		}
+		if err != nil {
+			return false
+		}
+	}
+	return true
+}
+
+// parseYaml is a helper method for parsing csds request yaml to nodematchers
+func parseYaml(path string, yamlStr string, nms *[]*envoy_type_matcher.NodeMatcher) error {
+	if path != "" {
+		// parse yaml to json
+		filename, err := filepath.Abs(path)
+		if err != nil {
+			return err
+		}
+		yamlFile, err := ioutil.ReadFile(filename)
+		if err != nil {
+			return err
+		}
+		js, err := yaml.YAMLToJSON(yamlFile)
+		if err != nil {
+			return err
+		}
+
+		// parse the json array to a map to iterate it
+		var data map[string]interface{}
+		if err = json.Unmarshal(js, &data); err != nil {
+			return err
+		}
+
+		// parse each json object to proto
+		for _, n := range data["node_matchers"].([]interface{}) {
+			x := &envoy_type_matcher.NodeMatcher{}
+
+			jsonString, err := json.Marshal(n)
+			if err != nil {
+				return err
+			}
+			if err = protojson.Unmarshal(jsonString, x); err != nil {
+				return err
+			}
+			*nms = append(*nms, x)
+		}
+	}
+	if yamlStr != "" {
+		var js []byte
+		var err error
+		// json input
+		if isJson(yamlStr) {
+			js = []byte(yamlStr)
+		} else {
+			// parse the yaml input into json
+			js, err = yaml.YAMLToJSON([]byte(yamlStr))
+			if err != nil {
+				return err
+			}
+		}
+
+		// parse the json array to a map to iterate it
+		var data map[string]interface{}
+		if err = json.Unmarshal(js, &data); err != nil {
+			return err
+		}
+
+		// parse each json object to proto
+		i := 0
+		for _, n := range data["node_matchers"].([]interface{}) {
+			x := &envoy_type_matcher.NodeMatcher{}
+
+			jsonString, err := json.Marshal(n)
+			if err != nil {
+				return err
+			}
+			if err = protojson.Unmarshal(jsonString, x); err != nil {
+				return err
+			}
+
+			// merge the proto with existing proto from request_file
+			if i < len(*nms) {
+				proto.Merge((*nms)[i], x)
+			} else {
+				*nms = append(*nms, x)
+			}
+			i++
+		}
+	}
+	return nil
+}
+
+// getValueByKeyFromNodeMatcher get value by key from metadata of nodematchers
+func getValueByKeyFromNodeMatcher(nms []*envoy_type_matcher.NodeMatcher, key string) string {
+	for _, nm := range nms {
+		for _, mt := range nm.NodeMetadatas {
+			for _, path := range mt.Path {
+				if path.GetKey() == key {
+					return mt.Value.GetStringMatch().GetExact()
+				}
+			}
+		}
+	}
+	return ""
+}
+
+// TypeResolver implements protoregistry.ExtensionTypeResolver and protoregistry.MessageTypeResolver to resolve google.protobuf.Any types
+type TypeResolver struct{}
+
+func (r *TypeResolver) FindMessageByName(message protoreflect.FullName) (protoreflect.MessageType, error) {
+	return nil, protoregistry.NotFound
+}
+
+// FindMessageByURL links the message type url to the specific message type
+// TODO: If there's other message type can be passed in google.protobuf.Any, the typeUrl and messageType need to be added to this method to make sure it can be parsed and output correctly
```

Comment:
wrap this comment around, it is very long in a single line.

Reply:
Just split it to two lines.